### ISSUE TITLE
subplot not working with -Jp polar projection

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#389]
**Description of the problem**
In subplot mode, the width or scale of each subplot should be omitted, since subplot will automatically calculate the subplot width and append the width to -J option. However, this doesn't always work for -Jp, since figure width is not always the last arguments.

**Full script that generated the error**

```
gmt begin Jp pdf
  gmt subplot begin 2x2 -Fs5c 
    gmt basemap -JPa5c/45 -R0/90/0/1 -Bxa45f -c1,1 
    gmt basemap -JPa5c/45 -R0/90/0/1 -Bxa45f -c1,2
    gmt basemap -JPa5c/45 -R0/90/0/1 -Bxa45f -c2,1
    gmt basemap -JPa5c/45 -R0/90/0/1 -Bxa45f -c2,2
  gmt subplot end
gmt end
```

**Full error message**

```
psbasemap [ERROR]: Syntax error -J option. Correct syntax:
	-Jp[a]<scale>[/<origin>][r] OR -JP[a]<width>[/<origin>][r]
	  <scale> is cm/units, or use <width> in cm
	  Optionally, prepend a for azimuths, append theta as origin [0],
	  or append r to reverse radial coordinates.
psbasemap [ERROR]: Offending option -JPa5c/45/1.9685i
```

**System information**

* Operating system: macOS
* Version of GMT: 6.0.0_32e95ef

**Comment by PaulWessel** [comment]
Hm, I wonder if a more robust implementation would be to specify a special symbol for the width or scale so we know where to replace it with the computed value.  Perhaps -JPa=/45 and the = is replaced by the computed width (-Jpa1.9685i/45)?

**Comment by seisman** [comment]
The syntax of -Jp projection `-JP[a]<width>/<origin>[r][z]` is a little weird to me. Most other projections put `<width>` as the last argument, while `-JP` as the first one. Prepending `a` and appending `r` or `z` make it more complicated.

**Comment by PaulWessel** [comment]
True, that is a weird syntax.  What idiot came up with that?  Oh, me...
So if we switched it in a backwards compatible way to
**-JP**[**a**]_width_[**+o**_origin_][**+r**][**+z**]
or something then it would be the last argument (once modifiers are recognized).  We cannot just switch width and origin in the old syntax since a/b looks the same as b/a.

**Comment by seisman** [comment]
> Hm, I wonder if a more robust implementation would be to specify a special symbol for the width or scale so we know where to replace it with the computed value. 

Seems a good solution.

> Perhaps -JPa=/45 and the = is replaced by the computed width (-Jpa1.9685i/45)?

I'm afraid `=` is not ideal if the long-format option #230 is implemented and merged.

**Comment by PaulWessel** [comment]
Right, so we want a symbol that is on most people's keyboards without accessing special options keys.  Perhaps the question mark (?).

**Comment by seisman** [comment]
Partially fixed by #421.

I modified the script above. It can generate a plot now, but the polar projections are distorted. 
```
gmt begin Jp png,pdf,ps
gmt subplot begin 2x2 -Fs5c
gmt basemap -JPa?/45 -R0/90/0/1 -Bxa45f -c1,1
gmt basemap -JPa?/45 -R0/90/0/1 -Bxa45f -c1,2
gmt basemap -JPa?/45 -R0/90/0/1 -Bxa45f -c2,1
gmt basemap -JPa?/45 -R0/90/0/1 -Bxa45f -c2,2
gmt subplot end
gmt end
```
Output:
![image](https://user-images.githubusercontent.com/3974108/52570877-6f057c00-2de2-11e9-984e-719cff0ac790.png)
